api/src/main/java/com/spotify/ffwd/output/OutputPlugin.java

```diff
@@ -104,6 +110,20 @@ protected void configure() {
                 bind(wrapKey).to(sinkKey);
                 expose(wrapKey);
             }
+
+            @Provides
+            @Singleton
+            public BatchingStatistics batchingStatisticsProvider(
+                CoreStatistics statistics, @Named("pluginId") String pluginId, Logger log
+            ) {
+                if (batching.isReportStatistics()) {
+                    log.info("Enabling reporting of batching-specific metrics for output " +
+                        input.getTypeLiteral().getRawType().getName());
+                    return statistics.newBatching(pluginId);
+                } else {
+                    return new NoopCoreStatistics().newBatching(pluginId);
```

Comment:
This is really only a nit, but the NOOP batching singleton could be exposed here so that the pluginId does not have to be provided.

Reply:
Gotcha :) A small change so I did it.